phoenix-core-client/src/main/java/org/apache/phoenix/jdbc/PhoenixStatement.java

```diff
@@ -623,9 +623,19 @@ public Pair<Integer, Tuple> call() throws SQLException {
                                     throw new UpgradeRequiredException();
                                 }
                                 state = connection.getMutationState();
-                                plan = stmt.compilePlan(PhoenixStatement.this, Sequence.ValueOp.VALIDATE_SEQUENCE);
                                 isUpsert = stmt instanceof ExecutableUpsertStatement;
                                 isDelete = stmt instanceof ExecutableDeleteStatement;
+                                if (isDelete && connection.getAutoCommit() &&
```

Comment:
The whole purpose of this is to identify if it is a ServerSelectDeleteMutation plan and then set an additional scan attribute. You can do this later too along with other checks for singleRowUpdate. You have the plan which has the statementcontext which has the scan. This way you don't have to introduce this specialized compilePlan method.